SWIG/linearalgebra.i

```diff
@@ -307,6 +315,17 @@ function(x,y) plot(as.data.frame(x)))
 %}
 #endif
 
+%{
+using QuantLib::Disposable;
+%}
+
+template <class T>
+class Disposable : public T {
```

Comment:
I'm kind of conflicted about exposing this.  It introduces a level of risk (and of caring about lifetimes) that's not usual in Python, and probably in the other languages as well.  The alternative would be to declare the methods here as returning simply `Array` and let the compiler figure out the automatic conversion in the wrappers.

Reply:
Actually I was thinking in the same direction and I've rewritten the interface definition to skip Disposable<Array> and added it to this PR. Some compilations fail on the build server due to "out of memory" exceptions. Can we increase the available memory on the build server?

Reply:
Ouch.  No, I don't think we have control of that.  I can try decreasing the optimization level and see if that gives the compiler more space to breathe.

Reply:
Is clang an option? clang is using approx. 40% less memory when compiling these interface files.

Reply:
I'm trying that for Python and it seemed to work.  R might be trickier.  I'll let you know.

Reply:
Using clang now.  Python 2 and 3 compile.  The R failure is a problem in the generated code.  I've raised an issue to SWIG and I'll put in a workaround for the time being...

Reply:
Cool, thanks for changing!